stdnum/ro/onrc.py

```diff
@@ -0,0 +1,78 @@
+# onrc.py - functions for handling Romanian ONRC numbers
+# coding: utf-8
+#
+# Copyright (C) 2020 Dimitrios Josef Moustos
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA
+
+from stdnum.exceptions import *
+from stdnum.util import clean, isdigits
+import datetime
+import re
+
+"""ONRC (Numarul de ordine din registrul comertului).
+
+The ONRC number is a 9-14 character string containing information
+about the type of company, county, sequence number, 4 char year,
+separated by /. The EUID identifier is based on this number.
+
+>>> validate('J52/750/22.11.2012')  # valid, strip date, leave year
+'J52/750/2012'
+
+>>> validate('X52/750/2012')  # invalid first (type) character
+Traceback (most recent call last):
+    ...
+>>> validate('J49/750/2012')  # first number is invalid county no
+Traceback (most recent call last):
+    ...
+>>> validate('J52/100000/2012') # second number too large
+    ...
+>>> validate('J52/750/3000') # last number (year) is in the future
+Traceback (most recent call last):"""
+
+# regular expression to check ONRC structure
+_onrc_re = re.compile(r'^[JFC](?P<county>[1-9]\d?)\/(?P<n>[1-9]\d{0,4})\/(?P<year>\d{4})$')
+_current_year=datetime.datetime.now().year
```

Comment:
Doing it this way means that the current year is stored on the module level which may cause problems for longer-running applications. Let's say you started the application that uses this code in December 2020 then in January 2021 numbers for 2021 would be considered invalid.

Reply:
Thank you for considering this case. My intention was speeding up by not asking for year every time, but you are right. Thank you for adding onrc number validation.